### ISSUE TITLE
Typical training results

### ISSUE THREAD

**Issue by edend10** [lucidrains/DALLE-pytorch#61]
Hi, great repo and thanks for sharing your work!

I'm trying your bird dataset example from the Colab with OpenAI's pretrained VAE. I wasn't able to get meaningful results so far on the Colab or on my own vm (Tesla T4 GPU). 

13 epochs in of `train_dalle.py` and only seeing these kinds of results:
![image](https://user-images.githubusercontent.com/7587607/109565035-e54fa680-7aaf-11eb-923d-a5281f6deca2.png)

On my vm I ran `$ python train_dalle.py --image_text_folder /parent/to/birds/dataset/directory` without changing any of the code (only replaced wandb with another experiment tracking framework, but I doubt that should make a difference)

Should the bird dataset work better with the pretrained VAE? Can you share some results or common training parameters/times/number of epochs?

**Comment by lucidrains** [comment]
@edend10 Hi Eden! Thanks for trying out the repository! I may have found a bug with the pretrained VAE wrapper, fixed in the latest commit https://github.com/lucidrains/DALLE-pytorch/blob/0.2.2/dalle_pytorch/vae.py#L82 :pray: I'll be training this myself this week, and ironing out any remaining issues (other than data and scale of course)

**Comment by edend10** [comment]
Thanks for the response @lucidrains !
Ohh interesting, I'll check out the changes and try it out. Will look out for more updates!

**Comment by CDitzel** [comment]
what are [those two ](https://github.com/lucidrains/DALLE-pytorch/blob/0985d852c7edacc21d93d23ceda5a3e1ed5351fa/dalle_pytorch/vae.py#L27)mapping functions for anyway?

Are they just for transforming the pixel value range for the input data they just over at OpenAI?

**Comment by AlexanderRayCarlson** [comment]
Hello! Thank you for this excellent work. I seem to be getting something similar - abstract sorts of blue squares when training in the colab notebook. It looks like the package (0.2.2) is updated with the latest fix - is there anything else needed to do at the moment?

**Comment by awilson9** [comment]
This is still happening for me as well on the pretrained VAE on 0.2.2

**Comment by liuqk3** [comment]
Hi @afiaka87, Amazing results! Can you share more details about your configurations? such as the dataset,  learning rate, lr scheduler? number of text and image (8192, right?) tokens? Thanks.

**Comment by afiaka87** [comment]
> Hi @afiaka87, Amazing results! Can you share more details about your configurations? such as the dataset,  learning rate, lr scheduler, number of text and image (8192, right?) tokens? Thanks.

I should mention the dataset I'm using includes images released by OpenAI with their DALL-E. The mannequin image is not being generated from text alone, it's from an image text pair. Anyway, my point is that my dataset is bad and I'm mostly just messing around. It's probably the case that using images generated from DALL-E itself is bound to converge much quicker than usual.

I'm using the defaults in train_dalle.py except for the BATCH SIZE and DEPTH. Pretrained OpenAI VAE, top_k=0.9, and `reversible=True`. I tried mixing attention layers but it adds memory.

I'm working on creating a hyperparameter sweep with wandb currently. I think a learning rate of 2e-4 might be better for depth greater than 12 or so.

I still can't get a stable learning rate with 64 depth.

**Comment by afiaka87** [comment]
"a professional high quality emoji of a spider starfish chimera . a spider imitating a starfish . a spider made of starfish . a professional emoji ."

![starfish_spider_chimera](https://user-images.githubusercontent.com/3994972/110666899-5584b900-818f-11eb-8a51-ae1bffe932ee.png)

Left it running at 16 dept, 12 layers, batch size of 12. The loss is going down at a steady consistent rate.

```json
  "DEPTH": {
      "desc": null,
      "value": 16
  },
  "HEADS": {
      "desc": null,
      "value": 8
  },
  "TOP_K": {
      "desc": null,
      "value": 0.85
  },
  "EPOCHS": {
      "desc": null,
      "value": 27
  },
  "SHUFFLE": {
      "desc": null,
      "value": true
  },
  "DIM_HEAD": {
      "desc": null,
      "value": 64
  },
  "MODEL_DIM": {
      "desc": null,
      "value": 512
  },
  "BATCH_SIZE": {
      "desc": null,
      "value": 12
  },
  "REVERSIBLE": {
      "value": true
  },
  "TEXT_SEQ_LEN": {
      "value": 256
  },
  "LEARNING_RATE": {
      "value": 0.0002
  },
  "GRAD_CLIP_NORM": {
      "value": 0.5
  }
```

**Comment by afiaka87** [comment]
Heres some more. The images filename is roughly similar to the phrase used to train. 
![toucan_whale_chimera](https://user-images.githubusercontent.com/3994972/110668686-2d965500-8191-11eb-8514-7156026f63c0.png)


![fox_in_a_field_at_night_painting](https://user-images.githubusercontent.com/3994972/110668543-08a1e200-8191-11eb-986b-948246885b6d.png)

![wildlife_western_sahara](https://user-images.githubusercontent.com/3994972/110668569-10618680-8191-11eb-801f-e22589e5bbed.png)
![crocker_amazon_san_fran_tall_vantage_point](https://user-images.githubusercontent.com/3994972/110668595-16effe00-8191-11eb-9f89-95809d17e633.png)

**Comment by tommy19970714** [comment]
@afiaka87  Thank you for sharing your report of Weights & Biases!
But I can't see the report because its project is private.
Can you allow us to see it?
<img width="1432" alt="スクリーンショット 2021-03-11 17 55 43" src="https://user-images.githubusercontent.com/14125841/110761666-5a329580-8293-11eb-9534-ca895436b03d.png">

**Comment by afiaka87** [comment]
Hm, does this work? 
https://wandb.ai/afiaka87/dalle_wandb_while_loop/reports/Training-on-OpenAI-DALL-E-Generated-Images--Vmlldzo1MTk1NDM

**Comment by afiaka87** [comment]
> Hi @afiaka87, Amazing results! Can you share more details about your configurations? such as the dataset, learning rate, lr scheduler, number of text and image (8192, right?) tokens? Thanks.

Just for more info on the dataset itself, it is roughly 1,100,000 image-text pairs that were generated by OpenAI's DALL-E. They presented roughly ~30k unique text prompts of which they posted the top 32 of 512 generations on https://openai.com/blog/dall-e/. Many images were corrupt, and not every prompt has a full 32 examples, but the total number of images winds up being about 1.1 million. If you look at many of the examples on that page, you'll see that DALL-E (in that form at least), can and will make mistakes. These mistakes are also in this dataset. Anyway I'm just messing around having fun training and what not. This is definitely not going to produce a good model or anything.

**Comment by afiaka87** [comment]
There are also a *large* number of images in the dataset which are intended to be used with the "mask" feature. I don't know if that's possible yet in DALLE-pytorch though.

**Comment by tommy19970714** [comment]
Thank you for shareing your result!
I will refer your parameters.

**Comment by afiaka87** [comment]
@tommy19970714 

I did a hyperparameter sweep with weights and biases. Forty Eight 1200 iteration runs of dalle-pytorch while varying Learning Rate, Depth and Heads, (minimizing the total loss at the end of each run). 

https://github.com/lucidrains/DALLE-pytorch/issues/84#issue-830997522

**Comment by afiaka87** [comment]
Most important thing to note here is that the learning rate actually needs to go up to about 0.0005 when dealing with ~26-32 depth

**Comment by afiaka87** [comment]
I've done a much longer training session on that same dataset here:

https://github.com/lucidrains/DALLE-pytorch/issues/86
